### PR TITLE
Adds missing tests for #7377

### PR DESCRIPTION
**Pull request by aninde**
### Context

The issue aggregates missing tests for issue https://github.com/handsontable/handsontable/issues/7377.

### How has this been tested?

I checked both tests with the Handsontable version `8.2.0` and with version `8.0.0` and `7.4.2`.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. https://github.com/handsontable/handsontable/issues/7377
2.
3.

### Checklist:

- [ ] My code follows the code style of this project,
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: src/plugins/hiddenColumns/test/editors.e2e.js [hunk 1]
```diff
@@ -36,4 +36,39 @@ describe('HiddenColumns', () => {
...
+      });
+
+      hot.selectCell(0, 1);
+      keyDown('enter'); // open the editor
+      await sleep(200);
+
+      document.activeElement.value = 'aa'; // type incorect value
+      keyDown('enter'); // confirm change the value
+      await sleep(200);
+      keyDown(Handsontable.helper.KEY_CODES.ESCAPE); // close the editor
```
Comment (swistach) [inline src/plugins/hiddenColumns/test/editors.e2e.js:67]: You can use `esc` instead.
`handsontableKeyTriggerFactory` will use proper keyCode internally: https://github.com/handsontable/handsontable/blob/dac8f5e68d8f21502183bb300e12395b36efa2be/test%2Fhelpers%2Fcommon.js#L390-L401

File: src/plugins/hiddenColumns/test/editors.e2e.js [hunk 2]
```diff
@@ -36,4 +36,39 @@ describe('HiddenColumns', () => {
...
+          { // the second numeric column
+            type: 'numeric',
+            allowInvalid: false,
+          },
+          {}, // the last column without validation
+        ]
+      });
+
+      hot.selectCell(0, 1);
+      keyDown('enter'); // open the editor
```
Comment (swistach) [inline src/plugins/hiddenColumns/test/editors.e2e.js:61]: I think `keyDownUp(...)` would be suited better to simulate a user's behaviour.

File: src/plugins/hiddenColumns/test/publicAPI.e2e.js [hunk 3]
```diff
@@ -125,7 +125,7 @@ describe('HiddenColumns', () => {
       });
 
       it('should return correct visual indexes when columns sequence is non-contiguous ' +
-         '(force desync between physical and visual indexes)', () => {
+        '(force desync between physical and visual indexes)', () => {
```
Comment (swistach) [inline src/plugins/hiddenColumns/test/publicAPI.e2e.js:128]: I guess your IDE did some auto-fixing here 🙃

File: src/plugins/hiddenColumns/test/publicAPI.e2e.js [hunk 4]
```diff
@@ -210,5 +210,31 @@ describe('HiddenColumns', () => {
         expect(emptyData).toEqual(empyDataComparision);
       });
     });
+
+    describe('Data change by setDataAtCell()', () => {
```
Comment (swistach) [inline src/plugins/hiddenColumns/test/publicAPI.e2e.js:214]: Just `setDataAtCell()` would be enough to describe what is inside (and would be consistent with the rest).

**Review by swistach — CHANGES_REQUESTED** [review]
Does `indicators` and `copyPasteEnabled` are necessary in `hidden*` configuration?

File: src/plugins/hiddenColumns/test/editors.e2e.js (hunk 2 above)
Comment (aninde) [reply src/plugins/hiddenColumns/test/editors.e2e.js]: Done by https://github.com/handsontable/handsontable/pull/7396/commits/741ea55cd3a3cc635ef4b7f68ee663f081f7a6d0.

File: src/plugins/hiddenColumns/test/publicAPI.e2e.js (hunk 3 above)
Comment (aninde) [reply src/plugins/hiddenColumns/test/publicAPI.e2e.js]: Done by https://github.com/handsontable/handsontable/pull/7396/commits/2d125a63c7c2b05e866b202b6e6b7ec847720f85.

File: src/plugins/hiddenColumns/test/publicAPI.e2e.js (hunk 4 above)
Comment (aninde) [reply src/plugins/hiddenColumns/test/publicAPI.e2e.js]: Done by https://github.com/handsontable/handsontable/pull/7396/commits/2d125a63c7c2b05e866b202b6e6b7ec847720f85.

**Comment by aninde** [comment]
I did not add similar tests to `hiddenRows` because this bug did not appear there.

**Review by swistach — APPROVED** [review]

**Pull request merged**